TFNetworkLayer.py

```diff
@@ -2970,14 +2981,15 @@ def _get_out_type_from_opts(cls, n_out, filter_size, padding, strides=1, dilatio
     padding = padding.upper()
     if input_expand_dims == 0 and not input_add_feature_dim and not input_split_feature_dim:
       # Maybe we have a chance to correctly define the output shapes.
-      data = get_concat_sources_data_template(sources)
+      index_shift = data.time_dim_axis_excluding_batch
       for i in range(len(filter_size)):
-        if data.shape[i] is not None:
-          shape[i] = cls.calc_out_dim(
-            in_dim=data.shape[i],
+        if data.shape[i + index_shift] is not None:
+          shape[i + index_shift] = cls.calc_out_dim(
+            in_dim=data.shape[i + index_shift],
             filter_size=filter_size[i], stride=strides[i], dilation_rate=dilation_rate[i], padding=padding)
-    feature_dim_axis = NotSpecified
-    if TFUtil.is_gpu_available() and False:  # TODO...
+    feature_dim_axis = data.feature_dim_axis
```

Comment:
This is wrong. It should be:
`feature_dim_axis = data.feature_dim_axis_or_unspecified`

Reply:
Actually, with my comment above, just leave it like it was (`feature_dim_axis = NotSpecified`).